ultravox_client/src/main/java/ai/ultravox/UltravoxSession.kt

```diff
@@ -24,6 +27,35 @@ class UltravoxSession(
     private val state = UltravoxSessionState()
     private var socket: WebSocket? = null
     private var room: Room = LiveKit.create(ctx)
+    private var speakerMuted: Boolean = false
+
+    val isMicMuted: Boolean
+        get() = !room.localParticipant.isMicrophoneEnabled()
+    
+    suspend fun setMicMuted(value: Boolean) {
+        room.localParticipant.setMicrophoneEnabled(!value)
+    }
+
+    suspend fun toggleMicMuted() {
+        setMicMuted(!isMicMuted)
+    }
+
+    val isSpeakerMuted: Boolean
+        get() = speakerMuted
+    
+    suspend fun setSpeakerMuted(value: Boolean) {
```

Comment:
1. Pretty sure this doesn't need suspend either, so you can directly make it the setter for speakerMuted.
2. A traditional for loop is probably better for readability than forEach with anonymous functions.